### ISSUE TITLE
"Update Manually" for some steps

### ISSUE THREAD

**Issue by pirotrav** [bluefin-octopusdeploy/chrome-extension#83]
Hello,

I use bluefin extension, and for the most part it has complemented Octopus really well.  However, recently I noticed an issue where certain steps templates inside a given project will not update via 'Update All' and instead say 'Update manually'.  The majority of them update, and it seems to most strongly impact step templates with lots of parameters, or possibly ones I pulled from the web (more likely the former than the latter is related).  The steps templates I wrote custom, not pulled from the library, do not use parameters and seem to update just fine.

We do a fair bit of cloning projects, importing projects etc which I suspect may be related.

Any explanation as to why this is happening, what to look for, and how to fix it?

Thanks,
Travis

**Comment by pirotrav** [comment]
I was able to verify the guid in the following message does exist to the project below:
Received message of updated process:deploymentprocess-Projects-1243

Action requires manual update: **d5a7bb35-6abb-4556-9da2-3175ad2b3d2d**

http://uat-30-octo01/api/deploymentprocesses/deploymentprocess-Projects-1243
"Id": "ba140c19-3105-4d48-8162-d95a981e9fe7",
      "Name": "IIS AppPool - Create",
      "RequiresPackagesToBeAcquired": false,
      "Properties": {
        "Octopus.Action.TargetRoles": "Integration-Web"
      },
      "Condition": "Success",
      "StartTrigger": "StartAfterPrevious",
      "Actions": [
        {
          "Id": "**d5a7bb35-6abb-4556-9da2-3175ad2b3d2d**",
          "Name": "IIS AppPool - Create",
          "ActionType": "Octopus.Script",
          "IsDisabled": false,
          "Environments": [],
          "Channels": [],
          "TenantTags": [],
          "Properties": {
            "Octopus.Action.Script.ScriptBody": 
... continued

**Comment by davidroberts63** [comment]
That feature will indicate a need for manual updating if the template
defines a new parameter with no default value and the in use step has no
property by that name defined. In that case Bluefin not be able to
determine an appropriate value for the new parameter.

Having said that Bluefin may be identifying some as needing manual updates
incorrectly. Can you provide the action properties you see? Appropriately
cleaned off any sensitive data of course.

**Comment by pirotrav** [comment]
Hi David,

Thanks for the reply and concise answer.  The step in question was one I downloaded from the Octopus community library (IIS apppool create).  It appears that there are several fields that are blank by default, which I had filled out in certain cases (IIS sites with a unique app pool identity and password) but not in other cases (default of application pool identity). The issue appears to be that when there is no user / password filled out, the tool says update manually as you implied.  I think this makes sense.

I became concerned since I had some issues with identities getting mixed up after moving projects between instances using the Octopus DB import / export tool.  Turns out you really need to be careful using that tool to move projects between Octopus instances, particularly with step references.  I'll have to go back to using the API or the octo.exe import / export utility.  Fortunately the issue here is much simpler, and makes sense now that I understand what's happening.

Thanks for your help, and maintaining such a useful tool!

Travis
